docs/utilities/config_annotations.md

```diff
@@ -0,0 +1,234 @@
+
+An example usage of the `@LangKey` is as follows:
+
+````Java
+@Config(modid = "my_mod_id")
+public class MyModConfig {
+
+  @Comment("This property has a non-default language key.")
+  @LangKey("my_mod_id.some_other_category.my_international_property")
+  public static boolean myInternationalProperty = false;
+
+  @LangKey("my_mod_id.some_other_category")
+  public static class MyCategory {
+
+      /* ... */
+
+  }
+
+}
+````
+
+!! Important
+
+    Applying this annotation to a category does not alter the language key of properties contained in that category.
+
+
+Time Specific Properties
+-------------------------
+
+Occasionally in order for a property change to take effect, a mod may require either a complete restart of Minecraft/Forge or a restart of the world. This distinction is important, as a property may control whether or not a sub-module is loaded or not, and a sudden change in the value of the property might cause the system to go looking for an unloaded sub-module. Ideally, the sub-module could be loaded on-the-fly, but in the case of a sub-module that adds a block to the game, attempting to load the sub-module would cause an error as blocks can only be added during initialization phases. Consequently, there exists a need as to signal the configuration system as to hold off making a change to the value specific properties in a `@Config` class until either a world reload occurs or a game reload occurs. For this reason, the annotations `@RequiresWorldRestart` and `@RequiresMcRestart` exist.
+
+The `@RequiresWorldRestart` annotation works by holding off applied changes to a configuration file until a world reload occurs, at which point the changes are then applied, and the world reloaded.
+
+The `@RequiresMcRestart` annotation works by saving the configuration change to disk but not updating the field after the initial load of the configuration occurs. In this manner the change will be reflected at the next launch, but will not affect the current run instance.
+
+An example usage of the two aforementioned annotations is as follows:
+
+````Java
+@Config(modid = "my_mod_id")
+public class MyModConfig {
+
+  @Comment("This property is always changeable!")
+  public static boolean myAlwaysChangeableProperty = false;
+
+  @Comment("This property is only changeable on world reloads!")
+  @RequiresWorldRestart
+  public static boolean myWorldReloadChangeableProperty = false;
+
+  @Comment("This property is only changeable on Minecraft restarts!")
+  @RequiresMcRestart
+  public static boolean myMcReloadChangeableProperty = false;
```

Comment:
This code snippet does not introduce any new information. A reader knows what an annotation is, how it is used, and what the phrases "world reload" and "minecraft restart" mean.